### PR TITLE
standardizing github workflows and consoldiating frontend check

### PR DESCRIPTION
**Pull request by CaioCintra**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Related issue

-  #431

### CODE REVIEW

**Comment by andrewtavis** [comment]
Thanks for the PR, @CaioCintra! We'll try to get to this in the coming days :) Appreciate you jumping in and helping us with this 😊

@to-sta, could you give this a first check?

**Comment by andrewtavis** [comment]
I left the maintainer checklist blank as well, @to-sta. Could you go through and check that? Figure it makes sense for you to go through the full process :)

File: .github/workflows/pr_ci_backend.yaml
Comment (to-sta) [inline .github/workflows/pr_ci_backend.yaml:1]: @CaioCintra thanks for your PR 😃. 

In the issue #407, I mentioned we should be more specific with the naming in our steps.

```yaml
- name: Run Black
  uses: psf/black@stable
  with:
    options: --check --verbose --exclude "migrations/"
    src: ./backend
- name: Run ruff
  run: ruff ./backend
- name: Run mypy - Static Type Checking
  run: mypy ./backend --config-file ./backend/mypy.ini
```
My suggestion would be:
```yaml
- name: Run Black - Formatting check
  uses: psf/black@stable
  with:
    options: --check --verbose --exclude "migrations/"
    src: ./backend
- name: Run ruff - Linting and import sorting check
  run: ruff ./backend
- name: Run mypy - Static type check
  run: mypy ./backend --config-file ./backend/mypy.ini
```

Another minor thing would be to change the job name from `python` to `backend`.

File: .github/workflows/pr_ci_frontend.yaml
Comment (to-sta) [inline .github/workflows/pr_ci_frontend.yaml:1]: We wanted to consolidate the frontend workflows `pr_ts_typecheck` and `pr_frontend_formatting_check` to `pr_ci_frontend` . As mentioned in the [issue](https://github.com/activist-org/activist/issues/431#issuecomment-1752686697).

Currently the install Prettier command will not work, since yarn is not installed. You need to install it before you use it. When you consolidate this workflow with pr_ts_typecheck than you should be good to go 😃.

```yaml
- name: Install Prettier
  run: yarn add prettier
```
Similar to above, I would say we change the job name from `prettier` to `frontend`.

**Review by to-sta — CHANGES_REQUESTED** [review]
@CaioCintra thanks  for PR. I only have a minor change request for the ci_backend.  The job name at  the moment is python. I think i would sense to switch it backend and for the frontend workflow to frontend.

**Comment by to-sta** [comment]
> I left the maintainer checklist blank as well, @to-sta. Could you go through and check that? Figure it makes sense for you to go through the full process :)

@andrewtavis i think i am not able to edit the checkboxes in the maintainer checklist.

**Comment by andrewtavis** [comment]
I'll check into this and get back to you, @to-sta :) Let us know if the PR looks ok, and I'll make sure that you have those edit rights before merging 😊

File: .github/workflows/pr_ci_frontend.yaml
Comment (CaioCintra) [reply .github/workflows/pr_ci_frontend.yaml:1]: I'm a bit confused about installing yarn, how do i call pr_ts_typecheck to do this for me? I think everything else is fine

Comment (to-sta) [reply .github/workflows/pr_ci_frontend.yaml:1]: Ahh i see. Yeah we want one workflow called `pr_ci_frontend` that combines the steps from `pr_ts_typecheck` and `pr_frontend_formatting_check`.

Comment (CaioCintra) [reply .github/workflows/pr_ci_frontend.yaml:1]:
```
      - name: Yarn Install
        uses: borales/actions-yarn@v4
        with:
          cmd: install
          dir: "frontend"
```

this is in ps_ts_typecheck, i put the same thing at pr_ci_frontend before yarn is used, right?

File: .github/workflows/pr_ci_frontend.yaml
Comment (to-sta) [inline .github/workflows/pr_ci_frontend.yaml:35]: Looks good, @CaioCintra and it works 👍!  

But we also wanted to **consolidate/combine** the `pr_ts_typecheck` with `pr_frontend_formatting_check`.

To consolidate the two workflows you need to add `contine-on-error: true` to prettier step and below that add the typecheck step. Finally we can delete the `pr_ts_typecheck.yaml` file.
```yaml
  - name: Run Prettier
    working-directory: ./frontend
    run: |
      yarn prettier . --check --config ../.prettierrc --ignore-path ../.prettierignore
    continue-on-error: true
      
 - name: Run Type Check
    uses: borales/actions-yarn@v4
    with:
      cmd: nuxi typecheck
      dir: "frontend"
```

**Review by to-sta — CHANGES_REQUESTED** [review]

File: .github/workflows/pr_ci_frontend.yaml
Comment (CaioCintra) [reply .github/workflows/pr_ci_frontend.yaml:35]: Ok, just adding this to frontend and removing ts file, right? I'm sending another PR

Comment (CaioCintra) [reply .github/workflows/pr_ci_frontend.yaml:35]: I think I messed something out, is not working anymore

Comment (to-sta) [reply .github/workflows/pr_ci_frontend.yaml:35]: No, worries. I will give the workflow a try locally now and the checks failed due to other reason 😄

Comment (andrewtavis) [reply .github/workflows/pr_ci_frontend.yaml:35]: Hey @CaioCintra! No stress on the workflow errors. Looks like @to-sta and I might have introduced some errors in a large rewrite we did earlier today. We'll check this soon and get back to you!

Comment (andrewtavis) [reply .github/workflows/pr_ci_frontend.yaml:35]: @to-sta: a3bb23a has `yarn nuxi typecheck` up and running again :) We had just missed adding some indexes.

Comment (CaioCintra) [reply .github/workflows/pr_ci_frontend.yaml:35]: Ok, thanks

Comment (to-sta) [reply .github/workflows/pr_ci_frontend.yaml:35]: One issue that is preventing the workflows from working is the indentation. 

```yaml
 - name: Run Type Check
    uses: borales/actions-yarn@v4
    with:
      cmd: nuxi typecheck
      dir: "frontend"

# Should look like this
  - name: Run Type Check
    uses: borales/actions-yarn@v4
    with:
      cmd: nuxi typecheck
      dir: "frontend" 
```
In case you are using VScode i can recommend the [GitHub Actions](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-github-actions) extension. It will help you with issues like that.

**Review by andrewtavis — APPROVED** [review]
Hey there @CaioCintra! Checked in with @to-sta and it's looking good to us! I'll go ahead and fix the indentation issues really quick :) Appreciate the work here! 🚀

**Pull request merged**
